examples/blocked_cholesky_gpu.py

```diff
@@ -125,43 +126,43 @@ def cholesky_blocked_inplace(a):
     gemm2 = TaskSpace("gemm2")        # Inter-block GEMM
     solve = TaskSpace("solve")        # Triangular solve
 
-    for j in range(a.shape[0]):
+    for j in range(len(a)):
         for k in range(j):
             # Inter - block GEMM
             @spawn(gemm1[j, k], [solve[j, k]], placement=loc)
             def t1():
-                out = clone_here(a[j,j])  # Move data to the current device
-                rhs = clone_here(a[j,k])
+                out = a[j, j]
+                rhs = a[j, k]
                 out = update(rhs, rhs, out)
-                copy(a[j,j], out)  # Move the result to the global array
+                a[j, j] = out
 
         # Cholesky on block
         @spawn(subcholesky[j], [gemm1[j, 0:j]], placement=loc)
         def t2():
-            dblock = clone_here(a[j, j])
+            dblock = a[j, j]
             dblock = cholesky(dblock)
-            copy(a[j, j], dblock)
+            a[j, j] = dblock
 
-        for i in range(j+1, a.shape[0]):
+        for i in range(j+1, len(a)):
             for k in range(j):
                 # Inter - block GEMM
                 @spawn(gemm2[i, j, k], [solve[j, k], solve[i, k]], placement=loc)
                 def t3():
-                    out = clone_here(a[i,j])  # Move data to the current device
-                    rhs1 = clone_here(a[i,k])
-                    rhs2 = clone_here(a[j,k])
+                    out = a[i, j]
```

Comment:
Just to confirm that I've understood: are we using the technique of indexing blocks, then elements? I like that approach, but I can also see value in indices referring directly to the indices in the logical array that's being partitioned.

Reply:
See comment [here](https://github.com/ut-parla/Parla.py/pull/22/files#r606687763).